lbz/authz.py

```diff
@@ -25,60 +21,25 @@
 LIMITED_ALLOW = -1
 
 
-class Authorizer(metaclass=Singleton):
-    allow = {}
-    deny = {}
-    action = None
-    outcome = None
-    allowed_resource = None
-    denied_resource = None
-    expiration = None
-    iss = None
-
-    def __init__(self, resource=None):
-        self.resource = resource
-        self._permissions = NestedDict()
-
-    def __getitem__(self, y):
-        return self._permissions[y]
+class Authorizer:
+    def __init__(self, auth_jwt: str, resource_name: str, permission_name: str):
+        self.outcome = DENY
+        self.allowed_resource = None
+        self.denied_resource = None
 
-    def __str__(self):
-        return json.dumps({self.resource: self._permissions})
+        self.resource = resource_name
+        self.permission = permission_name
+        self._set_policy(auth_jwt)
 
     def __repr__(self):
-        return self.__str__()
-
-    def __contains__(self, *args, **kwargs):
-        return self._permissions.__contains__(*args, **kwargs)
-
-    def __len__(self):
-        return len(self._permissions)
-
-    def __iter__(self):
-        return self._permissions.__iter__()
-
-    def add_permission(self, permission_name, function):
-        self._permissions[function] = permission_name
+        return (
+            f"Authorizer(auth_jwt=<jwt>, resource_name='{self.resource}', "
+            f"permission_name='{self.permission}')"
+        )
 
-    def set_resource(self, resource_name):
-        self.resource = resource_name
+    def check_access(self):
+        self.outcome = DENY
 
-    def validate(self, function_name):
-        if function_name not in self._permissions:
-            raise NotAcceptable()
-        if self.expiration is None:
-            warnings.warn(
-                "EXPIRATION_KEY will be mandatory with 0.2 please upgrade Authz provider",
-                DeprecationWarning,
-            )
-        if self.iss is None:
-            warnings.warn(
-                "Lack ALLOWED_ISS is a security risk You should add it.",
-                SecurityRiskWarning,
-            )
-        elif self.iss and self.iss != ALLOWED_ISS:
-            raise PermissionDenied(f"{self.iss} is not allowed token issuer")
-        self.set_initial_state(function_name)
         if self.deny:
```

Comment:
This attribute is tricky for me to spot, because declared outside of ` __init__`.